src/guide/6-agent-birth-death/1-agent-birth-device.rst

```diff
@@ -1,3 +1,151 @@
 Agent Birth from Agent Functions
 ================================
 
+Agent Creation Overview
+-----------------------
+The agent creation interface usable in agent functions is only able to create a single agent per existing agent per iteration. 
+Additionally, the agent type and state being created must be earlier specified as part of the model description. If you need
+to create more than one agent for each existing agent, you should use the *TODO: link to host agent creation api*
+
+**Note:** *Agents created by agent functions do not exist until the next layer.*
+
+Enabling Agent Creation
+-----------------------
+
+To create agents from agent functions, you must specify the type of agent the function produces when defining your agent functions:
+
+.. tabs::
+  
+  .. code-tab:: python
+  
+    # The agent type 'example_agent' is set as the agent output type
+    agent_fn1_description.setAgentOutput(example_agent)
+
+  .. code-tab:: cpp
+
+    // The agent type 'example_agent' is set as the agent output type
+    agent_fn1_description.setAgentOutput(example_agent);
+
+Creating the New Agent
+----------------------
+
+When agent output has been enabled for an agent function, the ``FLAMEGPU->agent_out`` object will become available within agent
+function definitions. This can be used to initialise the properties of the newly created agent.
+
+Agent Output from Device:
+
+.. tabs::
+
+  .. code-tab:: cpp
+  
+      FLAMEGPU_AGENT_FUNCTION(ExampleAgentOutputFn, MsgNone, MsgNone) {
+          // The output agent's 'x' variable is set
+          FLAMEGPU->agent_out.setVariable<float>("x", 12.0f);
+  
+          // Other agent function code
+          ...
+      }
+
+Agent variables which are not manually set will be initialised with their default values.
+
+Conditional Agent Creation
```

Comment:
Agent output is always conditional, it's message output which is configurable.

If agent output was mandatory, the entire agent population would double every time the agent function was called. There will only be some rare cases where that's the intended behaviour.